content-services/content-sources-backend
# Fixes 3170: add option to set external hostname

## Summary

so we can have the repo.config file return the correct hostname for gpg keys

## Testing steps

1.  create a repo with a gpg key set, and  snapshotting enabled
2.  fetch your repos list to get uuids for the repo and latest snapshots:   GET  http://localhost:8000/api/content-sources/v1/repositories/
4.   fetch this url:  http://localhost:8000/api/content-sources/v1/repositories/60dd106b-d08e-436a-bf34-53180713d4bc/snapshots/8a68fc2d-c8eb-4046-8c56-df44603f8206/config.repo

replacing the UUIDs for the repo & snapshot UUID.  Notice the URL is likely localhost:8000.  now re-launch the server with:

`OPTIONS_EXTERNAL_HOST="https://foobar.com"  go run cmd/content-sources/main.go api`

Refetch the config.repo file, notice the host for the gpg key has changed to foobar.com


## Checklist
